src/main/java/unicash/model/commons/Amount.java

```diff
@@ -29,6 +71,58 @@ public static boolean isValidAmount(double amount) {
         return amount >= 0.00;
     }
 
+    /**
+     * Returns true if a given amount is a non-negative value.
+     */
+    public static boolean isPositiveAmount(double amount) {
+        return amount >= 0.00;
+    }
+
+    /**
+     * Returns true if a given amount is a non-negative value.
+     */
+    public static boolean isWithinRange(double amount) {
+        return (amount > Integer.MIN_VALUE) && (amount < Integer.MAX_VALUE);
+    }
```

Comment:
These checks don't really make sense to me, would the "range" of our amount already be `>= 0.00`? `isWithinRange` behavior does not line up with the method behavior too since it's supposed to be "non-negative" but you're checking for $-2^{31}$

Might I suggest collating these checks into a single `isWithinRange` that properly enforces the range of $[0.00, 2^{31} - 1]$

Reply:
I've edited this method to make more sense, there was an error in the documentation, thanks for noticing. 

About the collation of these checks and the below-mentioned changes to the Amount class, I suggest I let @lipwei1808 work on them instead? So that there won't be any duplication of work.

Reply:
Ah, but if @lipwei1808 was already leaving out the `Amount` constraint checks for you to handle, wouldn't it make more sense to make the change here?